flang/lib/Lower/PFTBuilder.cpp

```diff
@@ -1151,10 +1153,13 @@ struct SymbolDependenceDepth {
     if (sym.has<semantics::UseDetails>() ||
         sym.has<semantics::HostAssocDetails>() ||
         sym.has<semantics::NamelistDetails>() ||
+        sym.has<semantics::ModuleDetails>() ||
         sym.has<semantics::MiscDetails>()) {
-      // FIXME: do we want to do anything with any of these?
+      // FIXME: do we want to do anything with any of these?  Other syms?
       return 0;
     }
+    if (sym.has<semantics::DerivedTypeDetails>())
+      return 0; // FIXME: Can't do derived type analysis yet.
```

Comment:
Why is this one special?

Reply:
It's distinct from the Use -- Misc bunch, because those are **intended** to trigger early exits, pending further analysis.
By contrast, the expectation is that this function **should** eventually be enhanced to handle derived type objects.  This would be a good place for an _unimplemented_ assert, but doing that would cause test _pre-fir-tree02.f90_ to fail.  Perhaps there was a similar choice made with respect to _pre-fir-tree05.f90_?

Reply:
pre-fir-tree05.f90 was added by going through a window rather than entering by the front door.